src/couch_replicator_connection.erl

```diff
@@ -95,21 +95,25 @@ handle_call({create, URL, Worker}, From, State) ->
             couch_stats:increment_counter([couch_replicator, connection, creates]),
             true = ets:insert_new(
                 ?MODULE,
-                #connection{host=Host, port=Port, owner=Pid, worker=Worker}
+                #connection{host=Host, port=Port, owner=Pid, worker=Worker, mref=monitor(process, Pid)}
             ),
-            {reply, ok, incr_monitor(Pid, State)}
+            {reply, ok, State}
     end.
 
 
-handle_cast({relinquish, Owner, WorkerPid}, State) ->
+handle_cast({relinquish, WorkerPid}, State) ->
     couch_stats:increment_counter([couch_replicator, connection, relinquishes]),
     case ets:lookup(?MODULE, WorkerPid) of
         [Worker] ->
-            ets:insert(?MODULE, Worker#connection{owner=undefined});
+            case Worker#connection.mref of
+                MRef when is_reference(MRef) -> demonitor(MRef);
```

Comment:
Maybe add `[flush]`, it would clean the message queue from DOWN messages in case owner died after casting that. Mostly for cosmetic reasons